source/dub/recipe/packagerecipe.d

```diff
@@ -531,11 +531,24 @@ struct BuildSettingsTemplate {
 		dst.addSourceFiles(collectFiles(cSourcePaths, "*.{c,i}"));
 		auto sourceFiles = dst.sourceFiles.sort();
 
+		// exclude import paths outside the project root from collection
+		import std.algorithm : map, filter, startsWith;
+		import std.typecons : tuple;
+		import std.path : pathSplitter;
+
+		auto projectImportPath = importPaths
+			.byKeyValue
+			.map!(pair => tuple(pair.key, pair.value
+				.filter!(path => !path.pathSplitter.startsWith("..".only))
+				.array
+			))
+			.assocArray;
+
  		// collect import files and remove sources
 		import std.algorithm : copy, setDifference;
 
 		auto importFiles =
-			chain(collectFiles(importPaths, "*.{d,di}"), collectFiles(cImportPaths, "*.h"))
+			chain(collectFiles(projectImportPath, "*.{d,di}"), collectFiles(cImportPaths, "*.h"))
```

Comment:
don't forget to also do this for cImportPaths

Reply:
Um... why? Well OK, maybe let's just do it for all paths. I don't see why not. Let's see what breaks.

Reply:
oh no, you definitely want to be able to specify source files with `..`, since for sub-packages they might want to share some kind of info file across all sub-packages that for example just contain some compile time helpers.

Also not all globs in parent directory scan recursively, but only specify single files, where this kind of exclusion would seem really arbitrary to the user.